File: src/153-Find-Minimum-in-Rotated-Sorted-Array.java
// Time complexity: O(logn);
// Space complexity: O(1);

class Solution {
    public int findMin(int[] nums) {

        int left = 0;
        int right = nums.length;
        int min = nums[left];

        while(left <= right){
            int mid = left + (right - left)/2;

            // min = Math.min(min, nums[mid]);
            min = min > nums[mid] ? nums[mid] : min ;

            if(nums[left] < nums[mid]) left = left + 1;
            else right = right - 1;
        }

        return min;
    }

}

File: src/JavaScript-Solutions/153-Find-Minimum-In-Rotated-Sorted-Array.js
/**
 * @param {number[]} nums
 * @return {number}
 */


var findMin = function(nums) {
 let minimum = nums[0];
 let left = 0;
 let right = nums.length - 1;
 
 while(left <= right){
     const mid = left + Math.floor((right - left)/2);
     minimum = Math.min(minimum, nums[mid]);
     
     if(nums[mid] < nums[right]){
         right = mid - 1;
     }else{
         left = mid + 1;
     }
 }
 
 return minimum;
};